### PR TITLE
Add precert and referral endpoints

### PR DESCRIPTION
**Pull request by ankit819**
Also, updated the readme with new endpoints.
https://eligible.atlassian.net/browse/ENG-914

### CODE REVIEW

File: lib/eligible/api_resource.rb [hunk 1]
```diff
@@ -16,9 +16,16 @@ def self.require_param(value, name)
     end
 
     def self.send_request(method, url, api_key, params, required_param_name = nil)
-      require_param(params[required_param_name], required_param_name) unless required_param_name.nil?
+      unless required_param_name.nil?
```
Comment (ankitdimania-eligible) [inline lib/eligible/api_resource.rb]: `if required_param_name.present?`

**Comment by ankitdimania-eligible** [comment]
LGTM, please take care of minor comment.

File: lib/eligible.rb [hunk 2]
```diff
@@ -88,11 +91,19 @@ def self.direct_response?(params)
...
+  def self.api_key?(params)
+    [:api_key, 'api_key'].any? { |k| params.key?(k) }
+  end
+
   def self.request(method, url, api_key, params = {}, headers = {})
     api_key ||= @@api_key
     test = self.test
-    api_key = params[:api_key] if params.key?(:api_key)
-    test = params[:test] if params.key?(:test)
+    api_key = params[:api_key] || params['api_key'] if api_key?(params)
```
Comment (ankitdimania-eligible) [inline lib/eligible.rb]: how about https://github.com/junegunn/insensitive_hash

File: lib/eligible/api_resource.rb (hunk 1 above)
Comment (ankit819) [inline lib/eligible/api_resource.rb]: present is not available here.

**Comment by ankit819** [comment]
@ankitdimania-eligible CC is having few issues. I will refactor to fix those.

File: lib/eligible.rb (hunk 2 above)
Comment (ankit819) [inline lib/eligible.rb]: Does it make sense to add an external gem only for this?

Comment (kovyrin) [inline lib/eligible.rb]: No

On Monday, 14 March 2016, Ankit Choudhary <notifications@github.com> wrote:

> In lib/eligible.rb
> <https://github.com/eligible/eligible-ruby/pull/22#discussion_r55990462>:
>
> >    def self.request(method, url, api_key, params = {}, headers = {})
> >      api_key ||= @@api_key
> >      test = self.test
> > -    api_key = params[:api_key] if params.key?(:api_key)
> > -    test = params[:test] if params.key?(:test)
> > +    api_key = params[:api_key] || params['api_key'] if api_key?(params)
>
> Does it make sense to add an external gem only for this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eligible/eligible-ruby/pull/22/files#r55990462>.
>


-- 
Oleksiy Kovyrin
http://kovyrin.net/

File: lib/eligible/coverage_resource.rb [hunk 3]
```diff
@@ -19,5 +15,9 @@ def self.get_uri
     def self.post_uri
       fail NotImplementedError, "Please implement class method #{self}.post_uri"
     end
+
+    class << self
```
Comment (ankitdimania-eligible) [inline lib/eligible/coverage_resource.rb]: why not put everything in this block?

**Comment by ankitdimania-eligible** [comment]
LGTM

File: lib/eligible/coverage_resource.rb (hunk 3 above)
Comment (kovyrin) [inline lib/eligible/coverage_resource.rb]: it is much harder to figure out if a method is a class or an instance method in those cases. I'd recommend us not to do that.

**Comment by ankit819** [comment]
@kovyrin @ankitdimania-eligible Shall I merge this?

**Comment by kovyrin** [comment]
Go ahead

**Comment by ankitdimania-eligible** [comment]
merging. I need the changes for another api - received_pdf

**Pull request merged**